cmd/podman/common/specgen.go

```diff
@@ -771,3 +775,63 @@ func parseThrottleIOPsDevices(iopsDevices []string) (map[string]specs.LinuxThrot
 	}
 	return td, nil
 }
+
+func parseSecrets(secrets []string) ([]string, map[string]string, error) {
+	secretParseError := errors.New("error parsing secret")
+	var mount []string
+	envs := make(map[string]string)
+	for _, val := range secrets {
+		source := ""
+		secretType := "mount"
+		target := ""
+		envOnly := false
+		split := strings.Split(val, ",")
+
+		// --secret mysecret
+		if len(split) == 1 {
+			source = val
+			mount = append(mount, source)
+			continue
+		}
+		// --secret mysecret,opt=opt
+		maybeSource := strings.SplitN(split[0], "=", 2)
+		if len(maybeSource) == 1 {
+			source = split[0]
+			split = split[1:]
+		}
+		// TODO: implement other secret options
+		for _, val := range split {
+			kv := strings.SplitN(val, "=", 2)
+			switch kv[0] {
+			case "source":
+				source = kv[1]
+			case "type":
+				if kv[1] == "mount" || kv[1] == "env" {
+					secretType = kv[1]
+				}
+			case "target":
+				// Target is an environment variable only option
+				envOnly = true
```

Comment:
Why do I need envOnly, can I just check if target!=""?

Reply:
Why not support target=/PATH/TO/SECRET for mount types?

I don't see any target tests below.  I also don't see any failure tests.

Reply:
I was going to support that in another PR, along with UID, GID, and Mode for mounted secrets, since it touches on a lot of other stuff in the mount secret backend

Reply:
I'll add some failure tests though